app/controllers/activitypub/collections_controller.rb

```diff
@@ -33,9 +33,9 @@ def set_size
   def scope_for_collection
     case params[:id]
     when 'featured'
-      @account.statuses.permitted_for(@account, signed_request_account).tap do |scope|
```

Comment:
Pinned status is public or unlisted, so you don't have to search for private toots and mentions by `permitted_for`. However, `permitted_for` does not return status if the account is blocked, so we need to check if it is blocked.